lib/config/ChainConfigManager.ts

```diff
@@ -0,0 +1,293 @@
+  private static _reverseConfigs: { [routeType: string]: number[] };
+
+  /**
+   * Lazy load the full list
+   * Adds dependencies if they don't already exist
+   */
+  static get chainConfigsWithDependencies(): { [chainId: number]: ChainConfig } {
+    if (ChainConfigManager._chainConfigsWithDependencies) {
+      return ChainConfigManager._chainConfigsWithDependencies;
+    }
+    ChainConfigManager._chainConfigsWithDependencies = ChainConfigManager._chainConfigs;
+    for (const dependencyMapping in ChainConfigManager._routeDependencies) {
+      for (const dependency of ChainConfigManager._routeDependencies[dependencyMapping]) {
+        for (const chainId in ChainConfigManager._chainConfigs) {
+          let dependentPresent = false;
+          let dependencyPresent = false;
+          for (const supportedRoutingType of ChainConfigManager._chainConfigs[chainId].routingTypes) {
+            dependentPresent = dependentPresent || supportedRoutingType.routingType == dependencyMapping;
+            dependencyPresent = dependencyPresent || supportedRoutingType.routingType == dependency;
+          }
+          // If we have the dependent but not the dependency, add it
+          if (dependentPresent && !dependencyPresent) {
+            ChainConfigManager._chainConfigsWithDependencies[chainId].routingTypes.push({
+              routingType: dependency as RoutingType,
+            });
+          }
+        }
+      }
+    }
+    return ChainConfigManager._chainConfigsWithDependencies;
+  }
+
+  static get reverseConfigs(): { [routeType: string]: number[] } {
+    if (ChainConfigManager._reverseConfigs) {
+      return ChainConfigManager._reverseConfigs;
+    }
+    ChainConfigManager._reverseConfigs = {};
+    for (const chainId in ChainConfigManager.chainConfigsWithDependencies) {
+      for (const supportedRoutingType of ChainConfigManager.chainConfigsWithDependencies[chainId].routingTypes) {
+        if (!ChainConfigManager._reverseConfigs[supportedRoutingType.routingType]) {
+          ChainConfigManager._reverseConfigs[supportedRoutingType.routingType] = [];
+        }
+        ChainConfigManager._reverseConfigs[supportedRoutingType.routingType].push(parseInt(chainId));
+      }
+    }
+    return ChainConfigManager._reverseConfigs;
+  }
+
+  /**
+   * @returns all ChainIds
+   */
+  public static getChainIds(): ChainId[] {
+    return Object.keys(ChainConfigManager.chainConfigsWithDependencies).map((c) => Number.parseInt(c));
+  }
+
+  /**
+   * @param routingType The RoutingType to check
+   * @returns all chains that support given RoutingType
+   */
+  public static getChainIdsByRoutingType(routingType: RoutingType): ChainId[] {
+    return ChainConfigManager.reverseConfigs[routingType] || [];
+  }
+
+  /**
+   * @param alarmEnabled Alarms set or not
+   * @returns all chains that have the given alarm setting
+   */
+  public static getChainIdsByAlarmSetting(alarmEnabled: boolean): ChainId[] {
```

Comment:
where would we need to get chains where alarm is not enabled? I wonder if something like `getAlarmedChainIds` which returns only ones with alarmEnabled = true would be cleaner

Reply:
We currently don't. I'll simplify it and we can always add the option in the future if we need it.